mesonbuild/scripts/depscan.py

```diff
@@ -34,9 +45,19 @@ def __init__(self, pickle_file: str, outfile: str, sources: T.List[str]):
         self.sources_with_exports = [] # type: T.List[str]
     
     def scan_file(self, fname: str) -> None:
-        for line in pathlib.Path(fname).read_text().split('\n'):
-            import_match = import_re.match(line)
-            export_match = export_re.match(line)
+        suffix = os.path.splitext(fname)[1][1:]
+        if suffix in lang_suffixes['fortran']:
+            self.scan_fortran_file(fname)
+        elif suffix in lang_suffixes['cpp']:
+            self.scan_fortran_file(fname)
+        else:
+            sys.exit('Can not scan files with suffix .{}.'.format(suffix))
```

Comment:
I think `raise SystemExit` is preferred to `sys.exit`.  However, in this case should it rather be `raise RuntimeError`?